test/e2e/framework.go

```diff
@@ -240,13 +301,20 @@ func (f *Framework) CreateNamespace(name string) {
 
 // DeleteNamespace deletes the namespace with the given name in the
 // Kubernetes API or fails the test if it encounters an error.
-func (f *Framework) DeleteNamespace(name string) {
+func (f *Framework) DeleteNamespace(name string, waitForDeletion bool) {
 	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: name,
 		},
 	}
 	require.NoError(f.t, f.Client.Delete(context.TODO(), ns))
+
+	if waitForDeletion {
```

Comment:
needed so we fully delete `projectcontour` ns between test suite runs

we could also combine everything into one suite, though that could get disorganized